src/api/Value.d.ts

```diff
@@ -0,0 +1,14 @@
+import TreePosition from '../core/TreePosition';
+
+/**
+ * Value represents a structure that may be converted to concrete data.
+ * @property treePosition - The position of the value in the tree.
+ * @property snapshot - A function that returns a string representation of the
+ * operation.
+ * @property isValue - A function that checks whether the given value is a value.
+ */
+export interface Value {
+  treePosition: TreePosition;
+  isValue(): boolean;
+  snapshot(): string;
```

Comment:
The snapshot function will return a different type for each Value:
- Strings should return string;
- Counter should return int;
- List should return an array;

We need that Value be generic on the type of snapshot, and that each Value implements it with a concrete type.